src/Disks/ObjectStorages/MetadataStorageFromPlainObjectStorage.cpp

```diff
@@ -215,6 +221,26 @@ void MetadataStorageFromPlainObjectStorageTransaction::removeDirectory(const std
     }
 }
 
+void MetadataStorageFromPlainObjectStorageTransaction::createHardLink(const std::string & path_from, const std::string & path_to)
+{
+    if (metadata_storage.object_storage->isWriteOnce())
```

Comment:
Question (not related to this PR), does it make sense to create `MetadataStorageFromPlainObjectStorage` for non-writeonce/readonly disks?

Reply:
Agree